cmd/resources/resource_utils.go

```diff
@@ -0,0 +1,122 @@
+package resources
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/iancoleman/strcase"
+)
+
+// we have certain tags that aren't a 1:1 match to their operation id names
+var mapTagToSchemaName = map[string]string{
+	"Access tokens":   "Tokens",
+	"Account members": "Members",
+	"Approvals":       "Approval requests",
+	"Code references": "Code refs",
+	"OAuth2 Clients":  "Oauth2 clients", // this is just so we don't kebab case to o-auth
+	"User settings":   "User flag settings",
+}
+
+func getResourceNames(name string) (string, string) {
+	if mappedName, ok := mapTagToSchemaName[name]; ok {
+		name = mappedName
+	}
+
+	resourceKey := strcase.ToKebab(name)
+	// the operationIds use "FeatureFlag" so we want to keep that whole, but the command should just be `flags`
+	if name == "Feature flags" {
+		resourceKey = "flags"
+	}
+	return name, resourceKey
+}
+
+func replaceMethodWithCmdUse(operationId string) string {
+	r := strings.NewReplacer(
+		"post", "create",
+		"patch", "update",
+		"put", "replace",
+	)
+
+	return r.Replace(operationId)
+}
+
+func removeResourceFromOperationId(resourceName, operationId string) string {
+	// operations use both singular (Team) and plural (Teams) resource names, whereas resource names are (usually) plural
+	var singularResourceName string
+	if string(resourceName[len(resourceName)-1]) == "s" {
```

Comment:
This could be easier to read with `strings.HasSuffix("teams", "s")` and then use `strings.TrimRight("teams", "s")`